Services/Currency.cs

```diff
@@ -0,0 +1,256 @@
+            "XAF Communauté Financière Africaine (BEAC) CFA Franc BEAC",
+            "XCD East Caribbean Dollar",
+            "XDR International Monetary Fund (IMF) Special Drawing Rights",
+            "XOF Communauté Financière Africaine (BCEAO) Franc",
+            "XPF Comptoirs Français du Pacifique (CFP) Franc",
+            "YER Yemen Rial",
+            "ZAR South Africa Rand",
+            "ZMW Zambia Kwacha",
+            "ZWD Zimbabwe Dollar"
+        };
+        #endregion
+        private static Lazy<Dictionary<string, string>> _currencyCodes =
+            new Lazy<Dictionary<string, string>>(delegate {
+                Dictionary<string, string> currencyCodes = new Dictionary<string, string>();
+                //the array above has code/description pairs, each in a string, like so:
+                // "CCC Description"
+                //The first 3 characters are the code
+                //then there is a single space
+                //then the description
+                foreach (string line in ISO4217) {
+                    string code = line.Substring(0, 3);
+                    string desc = line.Substring(4, line.Length - 4);
+                    currencyCodes.Add(code, desc);
+                }
+                return currencyCodes;
+            });
+
+
+        public static Dictionary<string, string> CurrencyCodes { get { return _currencyCodes.Value; } }
+
+        /// <summary>
+        /// Given a currency code, returns a string representing the currency.
+        /// </summary>
+        /// <param name="code">The currency code</param>
+        /// <returns>A string representing the currency code as symbol, or the same code if no symbol was found.</returns>
+        /// <example>Given "EUR", this will return "€".</example>
+        public static string GetCurrencySymbol(string code) {
+            //http://www.xe.com/symbols.php has a table of symbols
+            //TODO: implement symbols from that table
+            if (code == "EUR") {
+                return "€";
+            } else if (code == "USD") {
+                return "$";
+            }
+            else {
+                return code;
+            }
+        }
+
+        /// <summary>
+        /// Given a currency code, returns a string representing the format to be used to write currency amounts.
+        /// </summary>
+        /// <param name="code">The currency code</param>
+        /// <returns>A string representing the currency format</returns>
+        /// <example>Given "EUR", this will return "€".</example>
+        public static string GetCurrencyFormat(string code) {
+            //http://www.thefinancials.com/?SubSectionID=curformat has a table including formats
+            //TODO: implement symbols from that table
+            return "0.00";
+        }
+
+        public static string GetPriceAsString(double price, string code) {
+            return price.ToString(GetCurrencyFormat(code)) + " " + GetCurrencySymbol(code);
```

Comment:
This is wrong: the currency symbol is not always after the amount. For instance, $400.